experimental/tinystr_neo/src/ascii.rs

```diff
@@ -54,26 +55,218 @@ impl<const N: usize> TinyAsciiStr<N> {
         Ok(Self { bytes: out })
     }
 
+    #[inline]
     pub const fn from_str(s: &str) -> Result<Self, TinyStrError> {
         Self::from_bytes(s.as_bytes())
     }
 
+    #[inline]
     pub fn len(&self) -> usize {
-        self.bytes.iter().position(|x| *x == 0).unwrap_or(N)
+        if N <= 4 {
+            Aligned4::from_bytes(&self.bytes).len()
+        } else if N <= 8 {
+            Aligned8::from_bytes(&self.bytes).len()
+        } else {
+            self.bytes.iter().position(|x| *x == 0).unwrap_or(N)
+        }
     }
 
+    #[inline]
     pub fn is_empty(&self) -> bool {
         self.bytes[0] == 0
     }
 
+    #[inline]
     pub fn as_bytes(&self) -> &[u8] {
         &self.bytes[0..self.len()]
     }
 
+    #[inline]
     pub fn all_bytes(&self) -> &[u8; N] {
         &self.bytes
     }
 
+    /// Checks if the value is composed of ASCII alphabetic characters:
+    ///
+    ///  * U+0041 'A' ..= U+005A 'Z', or
+    ///  * U+0061 'a' ..= U+007A 'z'.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tinystr_neo::TinyAsciiStr;
+    ///
+    /// let s1: TinyAsciiStr<4> = "Test".parse()
+    ///     .expect("Failed to parse.");
+    /// let s2: TinyAsciiStr<4> = "Te3t".parse()
+    ///     .expect("Failed to parse.");
+    ///
+    /// assert!(s1.is_ascii_alphabetic());
+    /// assert!(!s2.is_ascii_alphabetic());
+    /// ```
+    #[inline]
+    pub fn is_ascii_alphabetic(&self) -> bool {
+        if N <= 4 {
+            Aligned4::from_bytes(&self.bytes).is_ascii_alphabetic()
+        } else if N <= 8 {
+            Aligned8::from_bytes(&self.bytes).is_ascii_alphabetic()
+        } else {
+            self.as_bytes().iter().all(u8::is_ascii_alphabetic)
+        }
```

Comment:
Note: I did the fast path only for 4 and 8 because that's what most processors can handle in a single register and because 16-byte TinyStr values are fairly uncommon.  We can easily add the 16-byte versions if necessary in the future.